packages/openapi-generator/src/parser/schema.spec.ts

```diff
@@ -518,18 +524,141 @@ describe('parseSchema', () => {
             ]
           }
         ]
-      }
-    );
-  });
+      });
+    });
 
-  it('parses not schema', async () => {
-    const schema: OpenAPIV3.SchemaObject = {
-      not: { type: 'object' }
-    };
-    expect(parseSchema(schema, await createTestRefs(), defaultOptions)).toEqual(
-      {
+    it('parses not schema', async () => {
+      const schema: OpenAPIV3.SchemaObject = {
+        not: { type: 'object' }
+      };
+      expect(
+        parseSchema(schema, await createTestRefs(), defaultOptions)
+      ).toEqual({
         not: emptyObjectSchema
-      }
-    );
+      });
+    });
+
+    it('adds discriminator when discriminator has mapping', async () => {
+      const discriminator = {
+        propertyName: 'discriminatingProperty',
+        mapping: {
+          a: '#/components/schemas/SchemaA',
+          b: '#/components/schemas/SchemaB'
+        }
+      };
+      expect(
+        parseSchema(
+          {
+            oneOf: [
+              { $ref: '#/components/schemas/SchemaA' },
+              { $ref: '#/components/schemas/SchemaB' }
+            ],
+            discriminator
+          },
+          await createTestRefs({
+            schemas: {
+              SchemaA: { properties: { c: { type: 'string' } } },
+              SchemaB: { properties: { c: { type: 'integer' } } }
```

Comment:
I might not fully understand this test, but I think it should be:

```suggestion
              SchemaA: { properties: { discriminatingProperty: { type: 'string' }, c: { type: 'string' } } },
              SchemaB: { properties: { discriminatingProperty: { type: 'string' }, c: { type: 'integer' } } }
```

The discriminator has to be a property of all schemas.

> It is important that all the models mentioned below anyOf or oneOf contain the property that the discriminator specifies.

Don't ask me why this redundancy is necessary, IDK 😄 
I guess it would also be possible for it to have different types, e.g. a string in `SchemaA` and a number in `SchemaB`.

Unfortunately, after checking again, the Azure spec seems to simply use this discriminator incorrectly. If you look at it in the swagger editor you can see that it is simply ignored and the type of the `messages` array is incorrect 😬

Reply:
Actually, they fixed this in the [preview 2024-08-01](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/cognitiveservices/data-plane/AzureOpenAI/inference/preview/2024-08-01-preview/inference.json)

Reply:
Oh damn, good catch. I read this multiple times yesterday, but still ignored it ;) 
Unfortunately that makes things quite a bit more complicated, but let's see.

Reply:
ok, it doesn't make it more complicated. types should still work as expected.